tests/js/sentry-test/performance/utils.ts

```diff
@@ -0,0 +1,169 @@
+import {RawSpanType} from 'sentry/components/events/interfaces/spans/types';
+import {EntryType, EventOrGroupType, EventTransaction} from 'sentry/types';
+
+import {PerformanceDetectorData} from './../../../../static/app/types/event';
+
+export enum ProblemSpan {
+  PARENT = 'parent',
+  OFFENDER = 'offender',
+}
+
+export const EXAMPLE_TRANSACTION_TITLE = '/api/0/transaction-test-endpoint/';
+
+type AddSpanOpts = {
+  endTimestamp: number;
+  startTimestamp: number;
+  childOpts?: AddSpanOpts[];
+  description?: string;
+  numSpans?: number;
+  op?: string;
+  parentSpanId?: string;
+  problemSpan?: ProblemSpan;
+  status?: string;
+};
+
+export class TransactionEventBuilder {
+  TRACE_ID = '8cbbc19c0f54447ab702f00263262726';
+  ROOT_SPAN_ID = '0000000000000000';
+  _event: EventTransaction;
+  _spans: RawSpanType[] = [];
+  _perfDetectorData: PerformanceDetectorData = {
+    causeSpanIds: [],
+    offenderSpanIds: [],
+    parentSpanIds: [],
+  };
+
+  constructor() {
```

Comment:
Should we allow overrides via the constructor. Say I want to have a different title, or a different id then default.
It would be neat if I could overide a certain property but keep everything else

Reply:
Good idea, yeah let's do that